### PR TITLE
fix(form-v2): use publicform context to selectively render components in form end page

### PR DESCRIPTION
**Pull request by tohhengxing**
## Problem

We should only show the form end page without other public form components such as sections and instructions.

Closes [#4453]

## Solution

public form context provider has property `submissionData` which is not null when form is submitted successfully. utilized that to selectively render components at form end page.

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

## Before & After Screenshots

**BEFORE**:

<details>

https://user-images.githubusercontent.com/102740235/182192377-49430351-6958-4189-a2e8-e2e3ee40a536.mp4
</details>

**AFTER**:

<details>

https://user-images.githubusercontent.com/102740235/182192398-16320d6f-65cc-4f7c-a31e-eacfa5dcafd8.mp4
</details>

### CODE REVIEW

File: frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx [hunk 1]
```diff
@@ -21,11 +21,12 @@ import { useFormSections } from '../FormFields/FormSectionsContext'
 
 import { SidebarLink } from './SidebarLink'
 
-export const SectionSidebar = (): JSX.Element => {
+export const SectionSidebar = (): JSX.Element | null => {
```
Comment (timotheeg) [inline frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx]: Why is this needed? It doesn't look like the changeset can make `SectionSidebar()` return `null`

**Review by timotheeg — COMMENTED** [review]
Thanks for another contribution @tohhengxing

File: frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx (hunk 1 above)
Comment (tohhengxing) [reply frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx]: Oops that was when I used null initially and I forgot to remove it. Sincere apologies @timotheeg

Comment (timotheeg) [reply frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx]: No worries at all. Remove it in a new commit and we can merge!

Comment (tohhengxing) [reply frontend/src/features/public-form/components/SectionSidebar/SectionSidebar.tsx]: removed! thank you very much!

**Review by timotheeg — APPROVED** [review]
